src/extension/reactDirManager.ts

```diff
@@ -12,14 +14,18 @@ import * as path from "path";
 export class ReactDirManager implements vscode.Disposable {
     public static ReactDirPath = path.join(vscode.workspace.rootPath, ".vscode", ".react");
 
-    constructor() {
+    public create(): Q.Promise<void> {
         let fs = new FileSystem();
         /* if the folder exists, remove it, then recreate it */
-        fs.removePathRecursivelyAsync(ReactDirManager.ReactDirPath)
-            .done(() => fs.mkDir(ReactDirManager.ReactDirPath));
+        return fs.removePathRecursivelyAsync(ReactDirManager.ReactDirPath)
+            .then(() =>
+                fs.mkDir(ReactDirManager.ReactDirPath));
     }
 
     public dispose(): void {
-        new FileSystem().removePathRecursivelySync(ReactDirManager.ReactDirPath);
+        new EntryPoint(vscode.window.createOutputChannel("React-Native")).runFunction("extension.deleteTemporaryFolder",
+            "Couldn't delete the temporary folder ${ReactDirManager.ReactDirPath}",
+            () =>
+                new FileSystem().removePathRecursivelySync(ReactDirManager.ReactDirPath), /*errorsAreFatal*/ true);
```

Comment:
Is this actually a fatal error? We are trying to clean up, but it is not strictly speaking required